build/gulpfile.vscode.linux.js

```diff
@@ -15,12 +15,31 @@ const util = require('./lib/util');
 const packageJson = require('../package.json');
 const product = require('../product.json');
 
+const linuxPackageRevision = Math.floor(new Date().getTime() / 1000);
+
+const flatpakManifest = {
+	appId: 'com.microsoft.CodeOSS',
+	sdk: 'org.freedesktop.Sdk',
+	runtime: 'org.freedesktop.Sdk',
+	runtimeVersion: '1.4',
+	base: 'io.atom.electron.BaseApp',
+	baseFlatpakref: 'https://s3-us-west-2.amazonaws.com/electron-flatpak.endlessm.com/electron-base-app-master.flatpakref',
+	command: "code-oss",
+	symlinks: [
+		['/share/code-oss/code-oss', '/bin/code-oss'],
+	],
+	finishArgs: [
+		'--share=ipc', '--socket=x11',  // Allow showing X11 windows.
+		'--filesystem=home',            // Allow access to home directory.
+		'--device=dri',                 // Allow OpenGL rendering.
```

Comment:
Other finish args to consider
 - `--socket=pulseaudio` if the app or any app extensions need audio output
 - `--share=network` to install extensions using the network connection
 - `--talk-name=org.freedesktop.Notifications` if this even uses libnotify for desktop notifications

Lastly a bit weird, but adding `--filesystem=/tmp` will make sure you get the system tmp directory, which chromium/electron is using for its single check. https://github.com/electron/electron/blob/master/chromium_src/chrome/browser/process_singleton_posix.cc#L12